From e011ac8f6a714a24678dc3c90b68fca9de1a4a66 Mon Sep 17 00:00:00 2001
From: Sushma Kyasaralli Thimmappa <skyasara2@skyasara2-desktop.(none)>
Date: Thu, 9 Aug 2012 08:16:35 -0400
Subject: Dalvik: Correcting the size of the JIT method passed to VTune.

BZ: 51544

VTune data currently shows a large amount of "Unknown" for some of the benchmarks,
this was due to sending the method and the padding size to VTune. This change now sends
just the JITted method size, and with this more of the JIT code is resolved, reducing the
"Unknown" percentage.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I361ae883cc350a5eac0c511c21b141fddfb9303a
Orig-MCG-Change-Id: If0cb4fcb82774985d48981d88af2f46a62958e40
Signed-off-by: Sushma Kyasaralli Thimmappa <skyasara2@skyasara2-desktop.(none)>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/Frontend.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/Frontend.cpp b/vm/compiler/Frontend.cpp
index 7d0b810..9eb4a87 100644
--- a/vm/compiler/Frontend.cpp
+++ b/vm/compiler/Frontend.cpp
@@ -2235,7 +2235,7 @@ void SendTraceInfoToVTune(CompilationUnit *cUnit, JitTraceDescription *desc) {
     jitMethod.method_name = name_with_params;
     jitMethod.class_file_name = (char*)(cUnit->method->clazz->descriptor);
     jitMethod.method_load_address = cUnit->baseAddr;
-    jitMethod.method_size = cUnit->totalSize - cUnit->headerSize;
+    jitMethod.method_size = *(u2 *)((char *)cUnit->baseAddr - 4);
     jitMethod.source_file_name = (char*)(cUnit->method->clazz->sourceFile);
 
     //Send the trace load event to the VTune analyzer
-- 
1.7.4.1

